lib/filtrex/utils/encoder.ex

```diff
@@ -20,17 +20,33 @@ defmodule Filtrex.Utils.Encoder do
   """
   defmacro encoder(comparator, reverse_comparator, expression, values_function \\ {:&, [], [[{:&, [], [1]}]]}) do
     quote do
+      import Filtrex.Utils.Encoder
+
       def encode(condition = %{comparator: unquote(comparator), inverse: true}) do
         condition |> struct(inverse: false, comparator: unquote(reverse_comparator)) |> encode
       end
 
       def encode(%{column: column, comparator: unquote(comparator), value: value}) do
+        values = 
+          unquote(values_function).(value)
+          |> intersperse_column_refs(column)
+
         %Filtrex.Fragment{
-          expression: String.replace(unquote(expression), "column", column),
-          values: unquote(values_function).(value)
+          expression: String.replace(unquote(expression), "column", "?"),
+          values: values
         }
       end
     end
   end
 
+  def intersperse_column_refs(values, column) do
+    column = String.to_existing_atom(column)
+
+    [quote do: s.unquote(column)]
```

Comment:
Forgive my ignorance, but how do we actually reference the schema table by using `s` here. Is this something we are assuming from Ecto?

Reply:
The answer lies here:

https://github.com/rcdilorenzo/filtrex/blob/c6980d9c5796e3a10f1d2947b4e0830926ce30b7/lib/filtrex/ast.ex#L11

This is where Filtrex assigns the primary table in the query to the alias `s`.

Reply:
Ah, I hate when I forget code that I wrote myself. Thanks for being patient with me. :slightly_smiling_face: